crates/fuel-core/src/service/adapters/sync.rs

```diff
@@ -81,7 +78,45 @@ impl PeerToPeerPort for P2PAdapter {
                 .get_transactions_from_peer(peer_id.into(), block)
                 .await
         } else {
-            Ok(None)
+            Err(anyhow::anyhow!("No P2P service available"))
+        }
+    }
+
+    async fn report_peer(
+        &self,
+        peer: PeerId,
+        report: PeerReportReason,
+    ) -> anyhow::Result<()> {
+        if let Some(service) = &self.service {
+            let service_name = "Sync";
+            let new_peer_report_reason: NewPeerReportReason = report.into();
+            service.report_peer(peer, new_peer_report_reason, service_name)?;
+            Ok(())
+        } else {
+            Err(anyhow::anyhow!("No P2P service available"))
+        }
+    }
+}
+
+struct NewPeerReportReason(PeerReportReason);
+
+impl From<PeerReportReason> for NewPeerReportReason {
+    fn from(reason: PeerReportReason) -> Self {
+        Self(reason)
+    }
+}
+
+impl PeerReport for NewPeerReportReason {
```

Comment:
These are arbitrary values. We should do a spike or something to determine good values. Should we move these to a config?

Reply:
Having a config struct or constants file containing all the reputation values seems useful. They could still be hardcoded for now, but it would let us manage all the reputation behaviors from a single place that's easy to find.

Reply:
Okay. I added a config to `P2PAdapter` it's just hardcoded in the place where we init the adapter, but it gives us options in the future.

Reply:
https://github.com/FuelLabs/fuel-core/issues/1340